02.calendar/cal.rb

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env ruby
+
+require 'date'
+require 'optparse'
+
+params = ARGV.getopts("y:m:")
+
+request_year = params["y"].to_i #入力した年
+request_month = params["m"].to_i #入力した月
+
+if request_year == 0 && request_month == 0
+  now = Date.today
+  this_year = now.year # 今年
+  this_month = now.month #今月
+  first_day = Date.new(now.year,now.month,1) #今月1日
+  last_day = Date.new(now.year,now.month,-1) #今月末日
+elsif request_year == 0 && request_month != 0
+  now = Date.today
+  this_year = now.year
+  first_day = Date.new(this_year,request_month,1)
+  last_day = Date.new(this_year,request_month,-1)
+else
+  first_day = Date.new(request_year,request_month,1)
+  last_day = Date.new(request_year,request_month,-1)
+end
+
+what_day = first_day.strftime('%a') # 1日の曜日
+
+# start_atはカレンダーで「1」が表示される位置
+if what_day == "Sun"
+  start_at = 0
+elsif what_day == "Mon"
+  start_at = 1
+elsif what_day == "Tue"
+  start_at = 2
+elsif what_day == "Wed"
+  start_at = 3
+elsif what_day == "Thu"
+  start_at = 4
+elsif what_day == "Fri"
+  start_at = 5
+else what_day == "Sat"
+  start_at = 6
+end
+
+days_num = last_day - first_day +1
+days = days_num.to_i #当月の日数
```

Comment:
直訳すると days_count_of_month とかかなー

あと first_day は 1 日であることが自明なので、
` - first_day + 1` = 0 となり、以下だけで良いかもです

```suggestion
days_count_of_month = last_day.day
```

Reply:
Date#dayを使って取得する形に変更しました。